src/Mod/Part/Gui/SectionCutting.cpp

```diff
@@ -147,6 +147,20 @@ SectionCut::SectionCut(QWidget* parent)
                 return;
             }
             BoundCompound = pcCompound->Shape.getBoundingBox();
+            // make parent objects of links visible to handle the case that
+            // the cutting is started when only an existing cut was visible
+            std::vector<App::DocumentObject*> compoundObjects;
+            pcCompound->Links.getLinks(compoundObjects);
+            for (auto itCompound = compoundObjects.begin(); itCompound != compoundObjects.end();

```

Comment:
Is there a reason not to use a range-based for loop here?

Reply:
Done now. Since this PR is meant to fix issues and to be backported to 0.20.x. I will make the general loop code modernization in #8252